homeassistant/components/mqtt/__init__.py

```diff
@@ -592,6 +605,13 @@ def _mqtt_on_subscribe(self, _mqttc, _userdata, mid, granted_qos):
 
     def _mqtt_on_message(self, _mqttc, _userdata, msg):
         """Message received callback."""
+        if msg.retain:
+            self.retained_messages[msg.topic] = (msg.payload, msg.qos)
```

Comment:
I don't know too much about Home Assistant's style preferences concerning usage of tuples. Creating a namedtuple/struct/... here would probably be unnecessary since it's only really used in a single other place.